### DISCUSSION TITLE
Is it possible to set an index on a column in a large list with PnP

### DISCUSSION THREAD

**Discussion by Keversap** [pnp/powershell#2876] [Q&A]
Hi,

I'm trying to set a column index via PowerShell on a column in a large list. I can do it manually by using the SharePoint interface but I'm not able to do it using code without getting an error about the list size. I'm using the code below which fails on the Executequery on a large list:

`        # get context`
`        $Context = Get-PnPContext     `
`        $fieldtocheck = Get-PnPField -List $list -Identity $field`
`        # set `
`        $fieldtocheck.indexed = $true`
`        $fieldtocheck.update()`
`        $Context.ExecuteQuery()`

Edit: exception:  
"ExecuteQuery" with "0" argument(s): "The attempted operation is prohibited because it exceeds the list view threshold."

**Comment by SinghGurd** [comment]
I have the exact same problem...Did you manage to solve this issue?

**Comment by Keversap** [comment]
Unfortunately not. I did notice that if I continued to query the same column via code, SharePoint would sometimes automatically place an index on the column. In my opinion this was too irregular. I had to solve it manually. A good reminder for future large lists to put the column index first.

**Comment by OASISfr** [comment]
Hello everybody,

i was searching for this too and have found solution :

`$field = Get-PnPField -List <LIST_NAME> -Identity '<FIELD_NAME>'`
`$field.EnableIndex()`
`$field.Context.ExecuteQuery()`
